subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/transform/ArtifactTransformBuildOperationIntegrationTest.groovy

```diff
@@ -0,0 +1,609 @@
+/*
+ * Copyright 2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.integtests.resolve.transform
+
+import groovy.transform.EqualsAndHashCode
+import org.gradle.api.internal.artifacts.transform.ExecuteScheduledTransformationStepBuildOperationType
+import org.gradle.integtests.fixtures.AbstractIntegrationSpec
+import org.gradle.integtests.fixtures.BuildOperationsFixture
+import org.gradle.internal.taskgraph.CalculateTaskGraphBuildOperationType
+import org.gradle.internal.taskgraph.CalculateTaskGraphBuildOperationType.PlannedNode
+import org.gradle.internal.taskgraph.NodeIdentity
+
+import java.util.function.Predicate
+
+class ArtifactTransformBuildOperationIntegrationTest extends AbstractIntegrationSpec implements ArtifactTransformTestFixture {
+
+    @EqualsAndHashCode
+    static class NodeId {
+        String nodeType
+        String nodeIdInType
+    }
+
+    static class NodeMatcher {
+        String phantomId
```

Comment:
```suggestion
        String nodeId
```

`phantomId` somewhat sounds strange to me. IIUC, this is the name/id assigned to the node if it matches the matcher. Maybe something like `assignedId` or even simpler `nodeId`/`id` would be better.

Reply:
Done